street2sat_utils/client.py

```diff
@@ -1,211 +1,199 @@
 
-
-def predict(images: List[np.ndarray]) -> List[str]:
-    torch.set_num_threads(1)
-    model = get_model()
-    all_results = []
-    with torch.no_grad():
-        for img in images:
-            all_results.append(run_prediction(img, model))
-    return all_results
-
-
-def run_prediction(img: np.ndarray, model: Model) -> str:
-    results = model(img)
-    results = results.pandas().xyxy[0].to_json(orient="records")
-    return results
-
-
-def get_model(path: str = "street2sat_utils/model_weights/best.pt") -> Model:
-    assert os.path.exists(path), "Model path does not exist!"
-    model = hubconf.custom(path)
-    model.eval()
-    return model
-
-
-def get_image(image_path: str) -> str:
-    img = cv2.imread(image_path)
-    img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
-    s = io.BytesIO()
-    fig = Figure()
-    ax = fig.subplots()
-    ax.imshow(img)
-    ax.axis("off")
-    fig.savefig(s, format="png")
-    return base64.b64encode(s.getbuffer()).decode("ascii")
-
-
-def plot_labels(
-    img: np.ndarray,
-    results: List[Dict[str, float]],
-    path_prefix: str = "street2sat_utils/crop_info/",
-) -> str:
-    classes = {}
-    with open(os.path.join(path_prefix, "classes.txt")) as classes_file:
-        for i, line in enumerate(classes_file):
-            classes[i] = line.strip()
-
-    img = np.copy(img)
-    colors = [
-        (0, 0, 255),
-        (0, 255, 0),
-        (255, 0, 0),
-        (0, 255, 255),
-        (255, 0, 255),
-        (255, 255, 0),
-    ]
-    all_classes_so_far: Dict[int, Tuple[int, int, int]] = {}
-    for dt in results:
-        t = int(dt["ymax"])
-        b = int(dt["ymin"])
-        l = int(dt["xmax"])
-        r = int(dt["xmin"])
-        c = int(dt["class"])
-
-        if c not in all_classes_so_far:
-            all_classes_so_far[c] = colors[len(all_classes_so_far.keys())]
-
-        cv2.rectangle(img, (l, t), (r, b), all_classes_so_far[c], 5)
-        cv2.putText(
-            img,
-            classes[c],
-            (r, b - 10),
-            cv2.FONT_HERSHEY_SIMPLEX,
-            2,
-            (0, 0, 0),
-            3,
-            cv2.LINE_AA,
+from constants import CROP_CLASSES, CROP_TO_HEIGHT_DICT, GOPRO_SENSOR_HEIGHT, MODEL_PATH
+from street2sat_utils import exif_utils
+
+model: Model = hubconf.custom(str(MODEL_PATH))
+model.eval()
+
+
+@dataclass
+class Prediction:
+    name: str
+    results: Dict
+    time: datetime
+    focal_length: int
+    coord: Tuple[int, int]
+    pixel_height: int
+    img: Optional[np.ndarray]
+    is_generate_distance: bool
+
+    def __post_init__(self):
+        if self.is_generate_distance:
+            self.distances = self.generate_distances()
+
+    @staticmethod
+    def load_img_from_bytes(img_bytes) -> np.ndarray:
+        img_np = np.frombuffer(img_bytes.read(), np.uint8)
+        img = cv2.imdecode(img_np, cv2.IMREAD_UNCHANGED)[:, :, ::-1]
+        img_bytes.seek(0)
+        return img
+
+    @classmethod
+    def from_results_and_tags(
```

Comment:
A prediction can be generated using several functions
- Prediction.from_img_path()
- Prediction.from_img_bytes()
- Prediction.from_results_and_tags()